Tapp 66: Create an application form component which renders under the route /application

dingtaoliu opened the pull request:
This form allows applicants to enter basic personal information and stores it in a persisted redux store.  Further features include saving the form information to the backend and creating a course selection interface.

chrislessard commented on frontend/src/modules/application/components/ApplicationForm.jsx:
```diff
@@ -1,11 +1,19 @@
     <input style={{ width: "30vw" }} value={curValue} onChange={onChange} required={required} />
 </div>
 
+// Textbox input field component
```
I think the name of the const is enough to explain this, personally. The comment feels a bit superfluous to me.

chrislessard commented on frontend/src/modules/application/components/ApplicationForm.jsx:
```diff
@@ -14,64 +22,65 @@ const TextboxInput = ({ label, curValue, onChange, required }) => <div>
 </div>
 
-const BinaryRadioInput = ({ label, curValue,required }) => <div>
+// Binary radio input field component
```
Same as above.

chrislessard commented on frontend/src/modules/application/components/ApplicationForm.jsx:
```diff
@@ -14,64 +22,65 @@ const TextboxInput = ({ label, curValue, onChange, required }) => <div>
-
-const initialState = allFields.reduce((acc, cur) => ({ ...acc, [cur.value]: "" }), {})
-
+// Application form component that holds multiple individual simple components defined earlier
```
I feel like this comment is trying to explain how the framework functions, which isn't really our responsibility. Not sure it's really necessary to keep.

chrislessard commented on frontend/src/modules/application/components/ApplicationForm.jsx:
```diff
@@ -14,64 +22,65 @@ const TextboxInput = ({ label, curValue, onChange, required }) => <div>
+                                "value": event.target.value})
+    }
 
+    // TODO: used for submitting the form and saving it in the backend
```
Im ok with leaving a todo in the comments, but can we make sure that an associated issue gets created for it ? That's easier for the team to track imo.

chrislessard commented on frontend/src/modules/application/components/ApplicationForm.jsx:
```diff
@@ -14,64 +22,65 @@ const TextboxInput = ({ label, curValue, onChange, required }) => <div>
 
-    render_field_type = ({ value, label, required, type }) => {
-        if (type === "textbox") {
+    // Renders a simple field component
```
Same comment as above.

chrislessard commented on frontend/src/modules/application/components/FormFields.jsx:
```diff
@@ -1,29 +1,44 @@
+// This file contains all of the fields for the application form.  
```
I like the explanation, but please convert this to a multiline comment for readability. `{/* A JSX comment */}`

chrislessard commented on frontend/src/modules/application/components/ApplicationForm.jsx:
```diff
@@ -0,0 +1,114 @@
+    [personalInformationFields, 
+    currentProgramInformationFields,
+    currentStatusFields, 
+    customQuestions])
```
If I understand your intentions correctly, you're trying to concatenate these various arrays into one ? You can simplify this code; concat allows you to achieve this by passing several parameters, i.e. `[].concat([1], [2],[3])`

chrislessard reviewed (changes requested):
This is great work, I have a few small nitpicks that you'll find in my comments. Can you rename your first two commits, and their associated messages, to be in line with our standard ? Admittedly this is partially my fault, since I was the one who asked if you could push your work early ;) 

Theres a number of bits on the comments you've included. Personally, I think the code (namely, the function/const names) speaks for itself in many of these situations. I don't think we should be explaining how react components work, or what they are achieving when the code can already do so, we should reserve our comments for why we make particular decisions that might not be so clear to a dev.

I know we've talked about the transience of the team, so maybe keeping very explicit comments might be helpful to people who are just starting out with React. I'm open to discussion about it. 

Otherwise, great work !

chrislessard commented:
@dingtaoliu I've also noticed that one of our routes does not appear to be working under your PR. I've attached a screenshot; you can reproduce by running your work and navigating to `http://localhost:8000/tapp/applicants` 

<img width="1014" alt="screen shot 2019-03-07 at 11 08 27 pm" src="https://user-images.githubusercontent.com/17230733/54007070-0df56d80-412e-11e9-8c0b-f42955ecefd7.png">

dingtaoliu committed:
feat(frontend): Created a ta application form component with minimal styling

The application form is currently accessible through the route '/application'.  It stores the state of the form fields
in a persisted redux store.  Currently this component does not handle submitting form fields to the api and storing it
on the backend database.

chrislessard commented:
LGTM !
